### ISSUE TITLE
`prefer-else-if` fails to flag `if` after `else if`

### ISSUE THREAD

**Issue by andersk** [sindresorhus/eslint-plugin-unicorn#3586]
This flags `prefer-else-if` on lines 2, 3, 4, and 5, as expected:

```js
if (a === 1) b = 1;
if (a === 2) b = 2; // prefer-else-if
if (a === 3) b = 3; // prefer-else-if
if (a === 4) b = 4; // prefer-else-if
if (a === 5) b = 5; // prefer-else-if
```

But if I fix it on only lines 2 and 4, the errors on line 3 and 5 unexpectedly disappear:

```js
if (a === 1) b = 1;
else if (a === 2) b = 2;
if (a === 3) b = 3;
else if (a === 4) b = 4;
if (a === 5) b = 5;
```
